Xamarin.Essentials/Calendar/Calendar.ios.cs

```diff
@@ -0,0 +1,127 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using EventKit;
+using Foundation;
+
+namespace Xamarin.Essentials
+{
+    public static partial class Calendar
+    {
+        static async Task<IEnumerable<DeviceCalendar>> PlatformGetCalendarsAsync()
+        {
+            await Permissions.RequireAsync(PermissionType.CalendarRead);
+
+            EKCalendar[] calendars;
+            try
+            {
+                calendars = CalendarRequest.Instance.Calendars;
+            }
+            catch (NullReferenceException ex)
+            {
+                throw new Exception($"iOS: Unexpected null reference exception {ex.Message}");
+            }
+            var calendarList = new List<DeviceCalendar>();
+
+            foreach (var t in calendars)
+            {
+                calendarList.Add(new DeviceCalendar
+                {
+                    Id = t.CalendarIdentifier,
+                    Name = t.Title
+                });
+            }
+            return calendarList;
+        }
+
+        static async Task<IEnumerable<DeviceEvent>> PlatformGetEventsAsync(string calendarId = null, DateTimeOffset? startDate = null, DateTimeOffset? endDate = null)
+        {
+            await Permissions.RequireAsync(PermissionType.CalendarRead);
+
+            var eventList = new List<DeviceEvent>();
+            var startDateToConvert = startDate ?? DateTimeOffset.Now.Add(defaultStartTimeFromNow);
+            var endDateToConvert = endDate ?? startDateToConvert.Add(defaultEndTimeFromStartTime);  // NOTE: 4 years is the maximum period that a iOS calendar events can search
+            var sDate = startDateToConvert.ToNSDate();
+            var eDate = endDateToConvert.ToNSDate();
+            EKCalendar[] calendars;
+            try
+            {
+                calendars = !string.IsNullOrWhiteSpace(calendarId)
+                    ? CalendarRequest.Instance.Calendars.Where(x => x.CalendarIdentifier == calendarId).ToArray()
+                    : null;
+            }
+            catch (NullReferenceException ex)
+            {
+                throw new NullReferenceException($"iOS: Unexpected null reference exception {ex.Message}");
+            }
+
+            var query = CalendarRequest.Instance.PredicateForEvents(sDate, eDate, calendars);
+            var events = CalendarRequest.Instance.EventsMatching(query);
+
+            foreach (var e in events)
```

Comment:
See previous comment on using linq

Reply:
Also, apply the sort condition as part of the linq expression

Reply:
Addressed in:
https://github.com/builttoroam/Essentials/pull/43